modules/util/args/BaseArgs.py

```diff
@@ -19,26 +19,25 @@ def to_args(self) -> str:
         for name in self.types:
             value = getattr(self, name)
             if value is not None:
-                if self.types[name] == str:
+                if self.types[name] is str:
```

Comment:
BaseArgs inherits from BaseConfig, which is a pretty much just a typed key/value dict with a few features
1. It can be loaded and saved to json
2. It can be converted to a UIState to be used in the UI
self.types saves the type of each key in the dict. Using isinstance instead doesn't work in all cases. One example is enums. They are stored as strings in the json file, but converted to actual instances of the enum when they are loaded. Also, some values are nullable. And calling isinstance doesn't work if the value is None.

Reply:
Alright. Sounds like a fair reason to use that. :)

Reply:
That sounds mostly like stuff that can be handled via `pydantic`.

Reply:
@hameerabbasi Yeah probably. Maybe a later refactoring.